src/main/java/org/voltdb/exportclient/KinesisFirehoseExportClient.java

```diff
@@ -0,0 +1,241 @@
+        public ListeningExecutorService getExecutor() {
+            return m_es;
+        }
+
+        public KinesisFirehoseExportDecoder(AdvertisedDataSource source)
+        {
+            super(source);
+
+            CSVStringDecoder.Builder builder = CSVStringDecoder.builder();
+            builder
+                .dateFormatter(Constants.ODBC_DATE_FORMAT_STRING)
+                .timeZone(m_timeZone)
+                .columnNames(source.columnNames)
+                .columnTypes(source.columnTypes)
+            ;
+            m_es = CoreUtils.getListeningSingleThreadExecutor(
+                    "Kinesis Firehose Export decoder for partition " + source.partitionId
+                    + " table " + source.tableName
+                    + " generation " + source.m_generation, CoreUtils.MEDIUM_STACK_SIZE);
+            m_decoder = builder.build();
+        }
+
+        private void validateStream() throws RestartBlockException, InterruptedException {
+            DescribeDeliveryStreamRequest describeHoseRequest = new DescribeDeliveryStreamRequest().
+                    withDeliveryStreamName(m_streamName);
+            DescribeDeliveryStreamResult  describeHoseResult = null;
+            String status = "UNDEFINED";
+            describeHoseResult = m_firehoseClient.describeDeliveryStream(describeHoseRequest);
+            status = describeHoseResult.getDeliveryStreamDescription().getDeliveryStreamStatus();
+            if(status.equalsIgnoreCase("ACTIVE")){
+                return;
+            }
+            else if(status.equalsIgnoreCase("CREATING")){
+                Thread.sleep(5000);
+                validateStream();
+            }
+            else {
+                LOG.error("Cannot use stream %s, responded with %s", m_streamName, status);
+                throw new RestartBlockException(true);
+            }
+        }
+
+        final void checkOnFirstRow() throws RestartBlockException {
+            if (!m_primed) try {
+                m_firehoseClient = new AmazonKinesisFirehoseClient(
+                        new BasicAWSCredentials(m_accessKey, m_secretKey));
+                m_firehoseClient.setRegion(m_region);
+                validateStream();
+            } catch (AmazonServiceException | InterruptedException e) {
+                LOG.error("Unable to instantiate a Amazon Kinesis Firehose client", e);
+                throw new RestartBlockException("Unable to instantiate a Amazon Kinesis Firehose client", e, true);
+            }
+            m_primed = true;
+        }
+
+        @Override
+        public boolean processRow(int rowSize, byte[] rowData) throws RestartBlockException
+        {
+            if (!m_primed) checkOnFirstRow();
+            Record record = new Record();
+            try {
+                final ExportRowData rd = decodeRow(rowData);
+                String decoded = m_decoder.decode(null, rd.values);
+                record.withData(ByteBuffer.wrap(decoded.getBytes()));
```

Comment:
why aren't you specifying the encoding in the getBytes() use StandardEncodings.UTF_8